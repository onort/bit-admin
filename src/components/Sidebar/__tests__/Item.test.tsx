import React from "react"
import { mount, shallow } from "enzyme"
import { MemoryRouter as Router, NavLink } from "react-router-dom"
import { MdAdd as AddIcon } from "react-icons/md"

import SidebarItem from "../Item"

describe("<SidebarItem />", () => {
  it("should match snapshot", () => {
    const wrapper = shallow(
      <Router>
        <SidebarItem icon={<AddIcon />} title="Add" to="/add" />
      </Router>
    )
    expect(wrapper.html()).toMatchSnapshot()
  })

  it("should have an icon, correct title and target url", () => {
    const title = "Add"
    const targetUrl = "/add"
    const wrapper = mount(
      <Router>
        <SidebarItem icon={<AddIcon />} title={title} to={targetUrl} />
      </Router>
    )
    expect(wrapper.find(AddIcon).exists()).toBe(true)
    expect(wrapper.find(".itemTitle").text()).toBe(title)
    expect(wrapper.find(`[href="${targetUrl}"]`).exists()).toBe(true)
    wrapper.unmount()
  })
})
