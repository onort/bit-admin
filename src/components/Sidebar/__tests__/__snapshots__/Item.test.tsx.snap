// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarItem /> should match snapshot 1`] = `"<a class=\\"item\\" href=\\"/add-bit\\"><span class=\\"itemIcon\\"><svg stroke=\\"currentColor\\" fill=\\"currentColor\\" stroke-width=\\"0\\" viewBox=\\"0 0 24 24\\" height=\\"1em\\" width=\\"1em\\"><path d=\\"M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z\\"></path></svg></span><span class=\\"itemTitle\\">Add</span></a>"`;
